### PR TITLE
Calls are now stored in a separate tree. 

### PR DESCRIPTION
**Pull request by happi**
More strict definition of tree objects.
Goes with epoch PR https://github.com/aeternity/epoch/pull/1064

### CODE REVIEW

File: contracts/contract_state_tree.md [hunk 1]
```diff
@@ -13,20 +14,51 @@ The contract state tree contains objects:
...
 - vm_version
 
+It is serialized as:
+```
+      CONTRACT_TYPE,
+      CONTRACT_VSN,
+      pubkey,
+      balance,
+      height,
+      nonce,
```
Comment (lucafavatella) [inline contracts/contract_state_tree.md]: I suggest deleting this paragraph from here and rather review https://github.com/aeternity/protocol/blob/b9b9eb2a707a18b7b976be78f3eca76e27c9c5e3/serializations.md#contract e.g. unclear (missing?) nonce there.

File: contracts/contract_state_tree.md [hunk 2]
```diff
@@ -13,20 +14,51 @@ The contract state tree contains objects:
...
 
-### Pruning of contracts
+The calls state tree contains all calls done in the block
+and their return values.
+
+- Calls & Return data
+```
+      	CONTRACT_INTERACTION_TYPE,
+      	CONTRACT_INTERACTION_VSN,
+      	caller_address,
```
Comment (lucafavatella) [inline contracts/contract_state_tree.md]: I suggest deleting this paragraph and reviewing https://github.com/aeternity/protocol/blob/b9b9eb2a707a18b7b976be78f3eca76e27c9c5e3/serializations.md#contract-call - there are two nonces there.

File: contracts/contract_state_tree.md (hunk 1 above)
Comment (happi) [reply contracts/contract_state_tree.md]: Hmm we need to describe both how a transaction is serizlized and how a state tree is serialized. Do we descibe both somewhere in protocol now?

Comment (happi) [reply contracts/contract_state_tree.md]: Aha, we do in that version.

File: contracts/contract_state_tree.md [hunk 3]
```diff
@@ -13,20 +14,23 @@ The contract state tree contains objects:
...
 
+### Pruning of contracts
+
+## Calls state tree
 
+The calls state tree contains all calls done in the block
+and their return values.
 
-### Pruning of contracts
+See [Call Serialization](../serializations.md#call) for the serialization specification.
```
Comment (lucafavatella) [inline contracts/contract_state_tree.md]: I do not see `#call` anchor. Maybe `#contract-call`?

File: contracts/contract_state_tree.md [hunk 4]
```diff
@@ -13,20 +14,23 @@ The contract state tree contains objects:
 - The contract log (data binary)
 - Active flag (Boolean)
 - List of refering contracts
-- Calls & Return data
 - Deposit (amount)
 - vm_version
 
-As well as normal account fields.
+See [Contract Serialization](../serializations.md#contract) for the serialization specification.
```
Comment (lucafavatella) [inline contracts/contract_state_tree.md]: It looks like a nonce is missing in the doc https://github.com/aeternity/epoch/blob/098fcfad68adc13fac7842103de299c20ffbc22d/apps/aecontract/src/aect_contracts.erl#L187

File: serializations.md [hunk 5]
```diff
@@ -245,6 +245,7 @@ subsequent sections divided by object.
 , <contract_address> :: binary()
```
Comment (lucafavatella) [inline serializations.md]: Is the nonce field expected here? it looks like there are two nonces here (while code has one https://github.com/aeternity/epoch/blob/098fcfad68adc13fac7842103de299c20ffbc22d/apps/aecontract/src/aect_call.erl#L133 ).

**Pull request merged**